FoodTracker/Controllers/MealTableViewController.swift

```diff
@@ -0,0 +1,163 @@
+            if let selectedIndexPath = tableView.indexPathForSelectedRow {
+                meals[selectedIndexPath.row] = meal
+                tableView.reloadRows(at: [selectedIndexPath], with: .none)
+            }else{
+                let NewIndexPath = IndexPath(row: meals.count, section: 0)
+                
+                meals.append(meal)
+                tableView.insertRows(at: [NewIndexPath], with: .automatic)
+            }
+            
+            saveMeals()
+        }
+    }
+    
+    //MARK: -Private Methods
+    private func loadData(){
+        
+        if let savedMeals = getMeals(){
+            meals += savedMeals
+        }else{
+            loadSampleMeals()
+        }
+    }
+    
+    private func loadSampleMeals(){
+        
+        let photo1 = UIImage(named: "meal1")
+        let photo2 = UIImage(named: "meal2")
+        let photo3 = UIImage(named: "meal3")
+        
+        guard let meal1 = Meal(mealName: "Fried Fish", mealRating: 4, mealImage: photo1) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        guard let meal2 = Meal(mealName: "Egg Fried Rice", mealRating: 3, mealImage: photo2) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        guard let meal3 = Meal(mealName: "Long Potato Chips", mealRating: 5, mealImage: photo3) else {
+            fatalError("Unable to initialize Meal1")
+        }
+        
+        meals += [meal1, meal2, meal3]
+        
+    }
+
+    private func getMeals() -> [Meal]?{
+        
+        return NSKeyedUnarchiver.unarchiveObject(withFile: Meal.ArchiveUrl.path) as? [Meal]
+    }
+    
+    public func saveMeals(){
+        
+        let isSuccessfulSave = NSKeyedArchiver.archiveRootObject(meals, toFile: Meal.ArchiveUrl.path)
+        
+        if isSuccessfulSave {
+            print("Meals saved successfully.")
+        }else{
+            print("Failed to save data.")
+        }
+    }
+}
+
+//MARK: -Extensions
+extension MealTableViewController: UITableViewDelegate{
+    
+    //MARK: -Edit Button methods
+    public func tableView(_ tableView: UITableView, commit editingStyle: UITableViewCell.EditingStyle, forRowAt indexPath: IndexPath) {
+        
+        if editingStyle == .delete {
+            meals.remove(at: indexPath.row)
+            saveMeals()
+            tableView.deleteRows(at: [indexPath], with: .fade)
+        }
+    }
+    
+    public func tableView(_ tableView: UITableView, canEditRowAt indexPath: IndexPath) -> Bool {
+        return true
+    }
+}
+
+extension MealTableViewController: UITableViewDataSource{
+    
+    public func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+
+    public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return meals.count
+    }
+
+    public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
```

Comment:
Always try to provide the restricted access to methods, properties, classes etc.